core/src/main/java/org/infinispan/factories/InterceptorChainFactory.java

```diff
@@ -255,6 +256,11 @@ public AsyncInterceptorChain buildInterceptorChain() {
                interceptorChain.appendInterceptor(createInterceptor(new ClusteredCacheLoaderInterceptor(), ClusteredCacheLoaderInterceptor.class), false);
             else
                interceptorChain.appendInterceptor(createInterceptor(new CacheLoaderInterceptor(), CacheLoaderInterceptor.class), false);
+
+            boolean transactional  = configuration.persistence().stores().stream().anyMatch(StoreConfiguration::transactional);
```

Comment:
@danberindei Good point! This also has implications for how the PersistenceManager determines which stores to write to when `!transactionMode.isTransactional()`, so I will have to make a few additional changes as well.